lisa/analysis/frequency.py

```diff
@@ -569,4 +573,34 @@ def plot_domain_frequency_transitions(self, pct: bool=False):
             )
         ).cols(1)
 
+    @TraceAnalysisBase.plot_method
+    @df_peripheral_clock_effective_rate.used_events
+    def plot_peripheral_frequency(self, clk_name, average: bool=True):
+        """
+        Plot frequency for the specified peripheral clock frequency
+
+        :param clk_name: The clock name for which to plot frequency
+        :type clk_name: str
+
+        :param average: If ``True``, add a horizontal line which is the
+            frequency average.
+        :type average: bool
+
+        """
+        logger = self.logger
+        df = self.df_peripheral_clock_effective_rate(clk_name)
+
+        freq = series_refit_index(df['effective_rate'], window=self.trace.window)
+        avg = series_mean(freq)
+        logger.info(
```

Comment:
Plot methods should be pure, aka only return a plot. If there is useful information that is non trivial to extract from the dataframes, it can be done in separate methods (or maybe even properties ?)

In this specific case I don't think it's really needed as:
1. the info is trivial to compute from the df
2. the tooltip for the HLine on the plot should provide the value (if someone is still using matplotlib backend, they really, really should move to bokeh backend to get interactive stuff like that)

Reply:
Okay, will drop the info.